radar-prmt-ios/auth/LoginViewController.swift

```diff
@@ -22,21 +22,23 @@ class LoginViewController: UIViewController {
             .subscribeOn(MainScheduler.instance)
             .subscribe(onSuccess: { [weak self] granted in
                 guard let self = self else { return }
-                if granted {
-                    self.performSegue(withIdentifier: "qrCodeScanner", sender: self)
-                } else {
-                    let alert = UIAlertController(title: "Camera access denied", message: "Camera permissions are turned off for the app. To scan a QR code, please enable these permissions.", preferredStyle: .actionSheet)
-                    if let settingsUrl = NSURL(string: UIApplication.openSettingsURLString) as URL? {
-                        alert.addAction(UIAlertAction(title: "Review permissions", style: .default, handler: { action in
-                            UIApplication.shared.open(settingsUrl, options: [:], completionHandler: nil)
-                            }))
-                    }
+                DispatchQueue.main.async {
```

Comment:
I thought this dispatch queue was already used due to the `subscribeOn(MainScheduelr.instance)`? Or does this solve the issue by using async? In that case, the `subscribeOn` method may not be needed.

Reply:
I don't know exactly what is happening when I remove `DispatchQueue.main.async`.
I got this error: `Terminating app due to uncaught exception 'NSInternalInconsistencyException', reason: 'threading violation: expected the main thread'`
And found this answer to this: `As per apple standards, you have to perform UI activity in the main queue only. DispatchQueue.main.async {}`

Reply:
I see I reversed `observeOn` and `subscribeOn`. If you change `subscribeOn` to `observeOn`, it should work without the additional async call: https://stackoverflow.com/a/58358985/574082